db/migrate/20210519204247_move_miq_set_memberships.rb

```diff
@@ -0,0 +1,65 @@
+class MoveMiqSetMemberships < ActiveRecord::Migration[5.0]
+  class Relationship < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class MiqSetMembership < ActiveRecord::Base
+  end
+
+  def up
+    say_with_time("Converting set relationships to miq_set_memberships") do
+      # The `ancestry` column holds the id for the `relationship` record that
+      # references the miq_set the parent record contains the miq_set object
+      #
+      # Because these are trivial trees (1 parent with 1 level of children),
+      # the `miq_set_id` can be derived from the `resource_id` of the parent
+      miq_set_id = "(SELECT rel2.resource_id FROM relationships rel2 WHERE rel2.id = rels.ancestry::bigint)"
+
+      connection.execute(<<-SQL.squish)
+        INSERT INTO miq_set_memberships (member_type, miq_set_id, member_id, created_at, updated_at)
+          SELECT rels.resource_type, #{miq_set_id}, rels.resource_id, rels.created_at, rels.updated_at
+          FROM relationships rels
+          WHERE rels.relationship = 'membership' AND rels.ancestry IS NOT NULL
+      SQL
+
+      Relationship.where(:relationship => "membership").delete_all
+    end
+  end
+
+  def down
+    say_with_time("Converting miq_set_memberships to set relationships") do
+      # Find the find the id for the `relationships` record that is from the
+      # 'membership' relationship, `miq_set_id` that matches the `resource_id`,
+      # and has a `ancestry` column value of NULL, all of which should be
+      # created in the first query.
+      ancestry_id = <<-ANCESTRY_ID.squish
+        (
+          SELECT rel2.id::varchar
+          FROM relationships rel2
+          WHERE rel2.relationship = 'membership'
+            AND rel2.resource_id = miq_set.miq_set_id
+            AND rel2.ancestry IS NULL
+        )
+      ANCESTRY_ID
+
+      # First Query:   Add root node
+      # Second Query:  Add child nodes
+      connection.execute(<<-SQL.squish)
+        INSERT INTO relationships (relationship, resource_type, resource_id, ancestry, created_at, updated_at)
+          SELECT 'membership', miq_set.member_type, miq_set.miq_set_id, NULL, miq_set.created_at, miq_set.updated_at
+          FROM miq_set_memberships miq_set
+          WHERE miq_set.miq_set_id IN (
+            SELECT miq_set_id
+            FROM miq_set_memberships miq_set_2
+            GROUP BY miq_set_2.miq_set_id
+          );
+
+        INSERT INTO relationships (relationship, resource_type, resource_id, ancestry, created_at, updated_at)
+          SELECT 'membership', miq_set.member_type, miq_set.member_id, #{ancestry_id}, miq_set.created_at, miq_set.updated_at
+          FROM miq_set_memberships miq_set;
+      SQL
```

Comment:
Can we have 2 separate execute statements for these?  Not sure why we need to mash them into a single execute.